streams-bootstrap-cli/src/test/java/com/bakdata/kafka/CliTest.java

```diff
@@ -400,4 +400,24 @@ public void run() {
                     .containsEntry("bar", "2");
         }
     }
+
+    @Test
+    void shouldProvideApplicationId() {
+        try (final KafkaStreamsApplication<?> app = new SimpleKafkaStreamsApplication<>(appId -> new SimpleCliStreamsApp(appId) {
+            @Override
+            public void buildTopology(final StreamsBuilderX builder) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public SerdeConfig defaultSerializationConfig() {
+                throw new UnsupportedOperationException();
+            }
+        })) {
+            KafkaApplication.startApplicationWithoutExit(app, new String[]{
+                    "--application-id", "application-1",
+            });
+            assertThat(app.getApplicationId()).isEqualTo("application-1");
```

Comment:
Please also test `app.createConfiguredApp().getUniqueAppId()` in order to improve test coverage (see Sonar check)